### PR TITLE
#27: Location: Earth service

### PR DESCRIPTION
**Pull request by afi13**
Issue #27

### CODE REVIEW

File: d8/modules/custom/dmaps/src/EarthGeographicalDistances.php [hunk 1]
```diff
@@ -0,0 +1,241 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\dmaps\EarthService.
```
Comment (ddrozdik) [inline d8/modules/custom/dmaps/src/EarthGeographicalDistances.php]: Please fix class name here.

File: d8/modules/custom/dmaps/src/EarthGeographicalDistances.php [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+class EarthGeographicalDistances {
+
+  /**
+   *  Radius of Earth semimajor.
+   */
+  const RADIUS_SEMIMAJOR = 6378137.0;
+
+  /**
+   * Earth flattening
```
Comment (ddrozdik) [inline d8/modules/custom/dmaps/src/EarthGeographicalDistances.php]: Please add point in the end of the comment.

File: d8/modules/custom/dmaps/src/EarthGeographicalDistances.php [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+  /**
+   * Convert longitude and latitude to earth-centered earth-fixed coordinates.
+   * X axis is 0 long, 0 lat; Y axis is 90 deg E; Z axis is north pole.
+   *
+   * @param $longitude
+   *   Longitude value.
+   * @param $latitude
+   *   Latitude value.
+   * @param int $altitude
```
Comment (DmitryDrozdik) [inline d8/modules/custom/dmaps/src/EarthGeographicalDistances.php]: Comment is absent for this variable.

File: d8/modules/custom/dmaps/src/EarthGeographicalDistances.php [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    $long = deg2rad($longitude);
+    $lat = deg2rad($latitude);
+    $radius = $this->getEarthRadius($latitude) * cos($lat);
+
+    if ($radius > 0) {
+      $angle = abs($distance / $radius);
+      $angle = min($angle, pi());
+    }
+    else {
+      $angle = pi();
```
Comment (ddrozdik) [inline d8/modules/custom/dmaps/src/EarthGeographicalDistances.php]: Use next construction:

``` php
$angle = pi();
if ($radius > 0) {
  $angle = abs($distance / $radius);
  $angle = min($angle, pi());
}
```

**Pull request merged**